sdk/ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -653,32 +652,36 @@ object WebSocketService {
           }
       }
       def dbQueries(
-          q: NonEmpty[Map[RPN, NonEmpty[Set[LfV]]]]
+          q: NonEmpty[Map[ContractTypeRef.Resolved, NonEmpty[Set[LfV]]]]
       )(implicit
           sjd: dbbackend.SupportedJdbcDriver.TC
       ): NonEmpty[Seq[(domain.ContractTypeId.Resolved, doobie.Fragment)]] =
-        q.toSeq map { case ((t, pn), lfvKeys) =>
+        q.toSeq flatMap { case (tid, lfvKeys) =>
           val keys = lfvKeys.toVector
           import dbbackend.Queries.joinFragment, com.daml.lf.crypto.Hash
-          (
-            t,
-            joinFragment(
-              keys map (k =>
-                keyEquality(
-                  Hash
-                    .assertHashContractKey(toLedgerApiValue(t), k, pn)
-                )
-              ),
-              sql" OR ",
-            ),
-          )
+          tid.allPkgIds.toSeq
+            .sortBy(_.toString)
+            .map(t =>
+              (
+                t,
+                joinFragment(
+                  keys map (k =>
+                    keyEquality(
+                      Hash
+                        .assertHashContractKey(toLedgerApiValue(t), k, tid.name)
+                    )
+                  ),
+                  sql" OR ",
+                ),
+              )
+            )
         }
 
       Future.successful(
         StreamPredicate[Positive](
           resolvedRequest.resolvedQuery,
           resolvedRequest.unresolved,
-          fn(resolvedRequest.q.map({ case (k, v) => (k._1, v) }).toMap.forgetNE),
+          fn(resolvedRequest.q.flatMap({ case (k, v) => k.allPkgIds.map(_ -> v) }).forgetNE.toMap),
```

Comment:
This use used to for tracking offsets per contract type. Returned events will contain a package-specific version of this, so we need to use that here too.